### PR TITLE
Switch to a Github action for CI testing

### PR DESCRIPTION
**Pull request by pgjones**
This is a basic initial version based on h11#89

### CODE REVIEW

**Comment by pgjones** [comment]
@Kriechi how does this look to you?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install tox
+      run: |
+        python -m pip install --upgrade tox
+    - name: Initialize tox envs
+      run: |
+        python -m tox --parallel auto --notest
+    - name: Test with tox
+      run: |
+        python -m tox --parallel 0
```
Comment (Kriechi) [inline .github/workflows/ci.yml]: Whats the advantage over calling tox as module via the python executable, vs. the tox binary directly?

File: src/wsproto/extensions.py [hunk 2]
```diff
@@ -278,6 +279,7 @@ def frame_outbound(
                 bits = self.client_max_window_bits
             else:
                 bits = self.server_max_window_bits
+            print(bits, self.client_max_window_bits, self.server_max_window_bits)
```
Comment (Kriechi) [inline src/wsproto/extensions.py]: Left over debug statements here and above?

**Comment by Kriechi** [comment]
Shouldn't this already have triggered Github Actions on this PR?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pgjones) [reply .github/workflows/ci.yml]: I think it was to ensure the installed version was found and used (I copied it). I don't think it is required and is less idiomatic, so now I've removed.

**Comment by pgjones** [comment]
> Shouldn't this already have triggered Github Actions on this PR?

I think it needs to be merged first, it has run [here](https://github.com/pgjones/wsproto/actions)

**Comment by Kriechi** [comment]
Looks good overall.

Thoughts on having an explicit lint and docs job?
Right now it is hard to find because it is part of the py38 job.

**Comment by pgjones** [comment]
> Thoughts on having an explicit lint and docs job?

Yep, any idea how to share common steps between jobs?

**Comment by pgjones** [comment]
Having read a bit more, including Hynek's article on github actions I think it is best to start like this (rather than splitting the jobs). It is fairly clear if it fails in any way.

**Pull request merged**